boot/bootutil/src/bootutil_misc.c

```diff
@@ -616,6 +642,15 @@ boot_set_pending(int permanent)
             rc = boot_write_image_ok(fap);
         }
 
+        if (rc == 0) {
```

Comment:
In BOOT_MAGIC_GOOD case swap_type should be checked and checked in case operation was interrupted here.

Reply:
I don't think anyone would be able to schedule a new swap operation if we were resuming an interruption; what exact issue do you have mind?

Reply:
I also don't think we need to worry about the resume case here.  `boot_set_pending()` doesn't get called from the boot loader, it gets called from the application.  If the application is running, there can't be a swap in progress.

This is a little weird, though.  If the application tries to schedule a swap operation, and one is already scheduled, we just return 0.  It seems like we ought to check if the two swap operations have the same type.  If they are different, we should return an error.  I think this should go in a separate PR, though.

Reply:
What I meant was a power-down had occurred just before writing the swap_type. Shouldn't an application be able to recognize such a undone request unambiguously, and be able to renew boot_set_pending() after that? For sure it is possible to check that the application had already collected proper upgrade image.

Reply:
@nvlsianpu I think I see what you mean.

Application schedules a "test" swap.  This requires two flash writes:
1. Write `magic` to secondary trailer.
2. Write `swap_type` to secondary trailer.

If the device reboots after step 1 (before step 2), the trailer will be incomplete.

However, I don't think this is a problem.  Since this is a new swap (not a resume), mcuboot won't even look at the `swap_type` field at startup.  It should execute the "test" swap as expected.

If mcuboot then reboots in the middle of this swap, then we have a potential issue.  Mcuboot will recognize that a swap was interrupted and needs to be resumed, but it won't be able to determine the swap type.  I think this just puts us back where we were before this bug fix ("revert" will fail; all other swap types happen to work by chance).  Since you cannot specify a "revert" swap type with the `boot_set_pending()` function, I don't think there is a problem here.

I do think there is a general issue of the device resetting with a partially-written trailer.  This issue has always existed, though, so it is probably best left for a separate PR.

Please let me know if I am still missing something.